src/test/java/com/amplitude/EventTest.java

```diff
@@ -0,0 +1,45 @@
+package com.amplitude;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+
+public class EventTest {
+  @Test
+  public void testCreateEventWithNullUserAndDeviceThrowsException() {
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> {
+          new Event("test event", null, null);
+        },
+        "Event must have one defined userId and/or deviceId");
+  }
+
+  @Test
+  public void testToJsonObject() {
+    Event event = new Event("test event", "test-user");
+    String longStr = "Long string to be truncated.";
+    for (int i = 0; i < 7; i++) {
+      longStr += longStr;
```

Comment:
Since this is the test, might be we don't need to use StringBuilder instead for improving the efficiency?

Reply:
+1 should use stringBuilder when building long strings, that's a good catch Alyssa. As an example, the Android tests are extremely long. Takes around two to three minutes to finish

Reply:
Hmm, for the test here, I guess the efficiency improvement is trivial though.. It doesn't hurt to change it, just won't be that much difference for limited loop.

Reply:
These strings are getting fairly long right, for the later truncate string test cases. Thanks for adding stringBuilder